fairseq/training.py

```diff
@@ -231,9 +244,28 @@ def create_fairseq_hydra_config(cls, fairseq_hydra_config, max_epoch, max_update
         }
         res.update(FairseqHydraConfig(config_dict, post_config_dict))
         return res
+    
+    def _fairseq_prepare_checkpoint(self, start_checkpoint):
+        # rename the start checkpoint to checkpoint_last.pt if it is not None and checkpoint_last.pt does not exist
+        if start_checkpoint is None:
+            print("No start checkpoint provided")
+            return
+        if not os.path.exists(start_checkpoint):
+            raise FileNotFoundError(f"Start checkpoint {start_checkpoint} does not exist")
+        if not os.path.exists(os.path.join(self.out_checkpoint_dir.get_path(), "checkpoint_last.pt")):
+            print(f"Linking {start_checkpoint} to {self.out_checkpoint_dir.get_path()}")
+            os.symlink(
+                start_checkpoint,
+                os.path.join(self.out_checkpoint_dir.get_path(), "checkpoint_last.pt")
+            )
+            os.symlink(
+                start_checkpoint,
+                os.path.join(self.out_checkpoint_dir.get_path(), os.path.basename(start_checkpoint))
+            )
```

Comment:
is this needed?

Reply:
I wonder if this can potentially break things depending on `start_checkpoint`. If its basename has a special meaning to fairseq (similar to `checkpoint_last.pt`), this could maybe result in unexpected behavior, right?

Reply:
We could maybe just assign a fixed name here like e.g. `checkpoint_initial.pt` (if that has no special role so far). Since it's a symlink, it's still very easy to see the original filename if that's of interest.